tests/beaker_tests/README-beaker-script-ref.md

```diff
@@ -0,0 +1,232 @@
+# Beaker Test Script Guide to Test Inputs
+
+#### Table of Contents
+
+* [Overview](#overview)
+* [Test Prerequisites](#tp)
+* [Return Codes](#rc)
+* [The `tests` Hash](#tests)
+ * [Top-level Keys](#tlk)
+ * [Test Case Keys](#tck)
+* [Title Pattern Testing](#tpt)
+
+## <a name="overview">Overview</a>
+
+The beaker test scripts have gone through a number of rewrites in an effort to reduce the time required to develop new test scripts. The current revision uses a `tests` hash as a container for all of the inputs required to fully run each test case. This document is a common reference for the `tests` hash and other required test variables.
+
+## <a name="tp">Test Prerequisites</a>
+
+Each test case in each test file makes basic assumptions about the state of the test environment:
+
+* The beaker host configuration file is set up with proper access credentials for agent and master
+* The Puppet Master daemon is running
+* SSH is enabled in the agent environment
+* The agent SSL certificate has been generated and signed by the Puppet Master
+
+## <a name="rc">Return Codes</a>
+
+The beaker tests will validate tests using return codes from bash and puppet commands:
+
+#### Bash Command Return Codes
+
+* `0` = successful command execution
+* `> 0` = unsuccessful command execution
+
+#### Puppet Command Return Codes
+
+* `0` = No Changes have occurred (idempotence tests)
+* `1` = An Error has been raised
+* `2` = Changes have occurred (manifest tests)
+* `4` = A Failure has occurred
+* `6` = A Change and Failure has occurred
+
+*Note: The `on()` method returns `0` as its default exit code*
```

Comment:
The default `:acceptable_exit_code` for the `on()` method is `0`.  An error will be thrown if the exit code does not match.  This can be customized using `:acceptable_exit_codes` in the options hash.